### PR TITLE
975 zoom based style guide

### PR DESCRIPTION
**Pull request by makella**
### Related to #973

Guide at staging:
* http://carto-vl-975-zoom-based-style-guide.developers.carto-staging.com/developers/carto-vl/guides/zoom-based-styling/

For the "General doc restructure" see #950
---

hiii @davidmanzanares can you please review this guide for me? I want to make sure that I covered everything you would have :)

I need to give it another read-through, add link URLs to the reference, etc. but I wanted to get your opinion on the content.

As a side note, since this guide is talking so much about zoom-based styling, is there a snippet somewhere that I can use to add the actual zoom level number on each map? I think that will help a lot.

@VictorVelarde once @davidmanzanares has done a review for content, and I work through any of his comments, I would love if you could take a look also and evaluate it from the perspective of the "voice" of the other guides that we have.

Also, is there some styling for an image inside of the guides? I have my .gif in the examples folder but didn't insert it in because I wasn't sure if we had a class for images.

thanks to both!!

### CODE REVIEW

**Comment by VictorVelarde** [comment]
Regarding to the zoom-level widget, we have a little "widget" in the editor/index.html. We could add something like that easily.

It would be like:
```js
        // Listen to every move event caused by the user
        const displayMapInfo = () => {
            const center = map.getCenter();
            const longitude = center.lng.toFixed(6);
            const latitude = center.lat.toFixed(6);
            const bearing = map.getBearing().toFixed(0);
            const zoom = map.getZoom().toFixed(2);
            const info = `Center: [${longitude}, ${latitude}] - Zoom: ${zoom} - Bearing: ${bearing}º`
            console.log(info);
            document.querySelector('.map-info').innerText = info;
        };
        map.on('move', displayMapInfo);
```
And then have a little panel under the map, like
```html
   <div id='map' class='map'></div>
   <div class="map-info"></div>
```

File: docs/guides/06-zoom-based-styling.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+
+We most often hear about zoom-based styling in the context of basemaps. As an example, if we look at one of CARTO’s basemaps, Voyager, across different zoom levels, we see that as the zoom level gets larger (zooming into the map), more features and labels are added. As these features and labels appear, their styling is adjusted to support the current view. 
+
+For example, major highways don’t display until zoom 5 and when they do, the appearance transitions from a single, small width line at low zoom levels to cased lines at larger zoom levels.
+
+[MULTI-SCALE GIF of Voyager]
+
+### Zoom-based styling with VL
+
+In the past, zoom-based styling has been overly complex creating a barrier of entry for many. It is no wonder that it was “easier” to put too much information on a map and/or only design a map for one zoom level.
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply docs/guides/06-zoom-based-styling.md]: I'm gonna add that to the intro

File: docs/guides/06-zoom-based-styling.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+To get started, copy and paste the code below into your favorite text editor and save it to a file `vancouver-trees.html`.
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+    <!-- Include CARTO VL JS -->
+    <script src="../../../dist/carto-vl.js"></script>
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply docs/guides/06-zoom-based-styling.md]: I'm gonna update this in-place, using the same references that in other guides. I'm also going to update the corresponding examples (step-1 to step-8)

**Comment by VictorVelarde** [comment]
I have included David's suggestions in your PR with my last commits and I'm deploying the guide to staging

**Comment by makella** [comment]
thanks so much for all the help @victor!! super appreciate

**Pull request merged**